Library/Homebrew/utils.rb

```diff
@@ -498,11 +498,19 @@ def pretty_ratelimit_reset(reset)
 
   class AuthenticationFailedError < Error
     def initialize(error)
-      super <<-EOS.undent
-        GitHub #{error}
-        HOMEBREW_GITHUB_API_TOKEN may be invalid or expired, check:
+      message = "Github #{error}\n"
+      if ENV["HOMEBREW_GITHUB_API_TOKEN"]
+        message << <<-EOS.undent
+          HOMEBREW_GITHUB_API_TOKEN may be invalid or expired; check:
           #{Tty.em}https://github.com/settings/tokens#{Tty.reset}
-      EOS
+        EOS
+      else
+        message << <<-EOS.undent
+          You may have old Github credentials saved in the OS X Keychain
```

Comment:
Actually thinking about it more: ideal case would be to just clear this automatically and retry if this fails on an API call to a public repo?